shm/Island.cpp

```diff
@@ -0,0 +1,20 @@
+#include "Island.hpp"
+#include "Coordinates.hpp"
+
+Island::Island() : position_(Coordinates(0, 0))
+{
+}
+Island::Island(int x, int y) : position_(Coordinates(x, y))
+{
+}
+
+std::ostream& operator<<(std::ostream& os, Island& island)
```

Comment:
Assume that `Island` is part of DLL which you provide to your customer. If it was defined as a `friend` of Island class, then "end user" who has DLL and .hpp file would know that there is << operator for Island class. In your case user won't have any idea about it.